pkg/controller/enterprisesearch/config.go

```diff
@@ -40,7 +41,12 @@ const (
 )
 
 func ConfigSecretVolume(ent entv1beta1.EnterpriseSearch) volume.SecretVolume {
-	return volume.NewSecretVolumeWithMountPath(name.Config(ent.Name), "config", ConfigMountPath)
+	return volume.NewSecretVolume(name.Config(ent.Name), "config", ConfigMountPath, ConfigFilename, 0644)
```

Comment:
If the config file already exists, entsearch does not attempt to modify it. So, should we be as restrictive as possible and set `0444` or even `0400`?

Reply:
`0444` sounds good, yes 👍 
I'm a bit afraid of `0400` because of the various ways the container may be ran with a different user.